bambi/backends/stan.py

```diff
@@ -69,9 +69,13 @@ def build(self, spec, reset=True):
         self.X['N'] = n_cases
 
         def _sanitize_name(name):
-            ''' Stan only allows alphanumeric chars and underscore, so replace
-            all invalid chars with '_' and track for later re-substitution. '''
-            clean = re.sub('[^a-zA-Z0-9\_]+', '_', name)
+            ''' Stan only allows alphanumeric chars and underscore, and
+            variable names must begin with a letter. So replace all invalid
+            chars with '_', prepend with 'b_', and track for later
+            re-substitution. '''
+            if name in self._original_names:
+                return name
+            clean = 'b_' + re.sub('[^a-zA-Z0-9\_]+', '_', name)
```

Comment:
Yeah, so the u_ and b_ prefixes are gone as far as the user can see. (Well, unless they look at the underlying Stan model code, I guess.) The issue is just that Stan doesn't like parameter names to start with numbers, but this conflicts with our random intercept terms that have names like '1|subject'. So I had a prepend some arbitrary letter prefix to avoid this, and chose 'b_' for no particular reason -- I guess my thought was just that if a user did somehow look at the underlying Stan code and saw the b_ prefixes mysteriously added to the parameter names, maybe they wouldn't think it looked too weird.